pkgs/applications/networking/instant-messengers/rambox/fetchNodeModules.nix

```diff
@@ -0,0 +1,26 @@
+{ stdenv, jq }: { src, nodejs, sha256 }:
+
+# Only npm >= 5.4.2 is deterministic, see:
+# https://github.com/npm/npm/issues/17979#issuecomment-332701215
+assert stdenv.lib.versionAtLeast nodejs.version "8.9.0";
+
+stdenv.mkDerivation {
+  name = "node_modules";
+
+  outputHashAlgo = "sha256";
+  outputHash = sha256;
+  outputHashMode = "recursive";
+
+  nativeBuildInputs = [ jq nodejs ];
+
+  buildCommand = ''
+    cp -r ${src}/* .
+    HOME=. npm install --force --ignore-scripts --only=production
+    for f in $(find node_modules -name package.json); do
+      # https://github.com/npm/npm/issues/10393
+      jq -S 'delpaths(keys | map(select(startswith("_")) | [.]))' $f > $f.tmp
+      mv $f.tmp $f
+    done
+    mv node_modules $out
```

Comment:
This snippet is actually pretty cool. We can use it to build other node.js applications without adding thousands of npm dependencies to nixpkgs.

Reply:
@Mic92 Thanks! I'll test it with another package or two and then I'll move it to `build-support`. Pretty sure it's deterministic now, but ignoring scripts (which are almost always non-deterministic) might be a problem.

I also have a fetcher for Maven/Leiningen projects: https://raw.githubusercontent.com/yegortimoshenko/overlay/master/pkgs/fetchMaven/default.nix